packages/api/cms-api/src/auth/auth.module.ts

```diff
@@ -18,9 +33,21 @@ export interface AuthModuleAuthedUserOptions<CurrentUser extends CurrentUserInte
     currentUserDto: Type<CurrentUser>;
 }
 
+export interface AuthModuleJwtOptions<CurrentUser extends CurrentUserInterface> extends Pick<ModuleMetadata, "imports"> {
+    readonly strategy: "jwt";
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    useFactory: (...args: any[]) => Promise<AuthModuleJwtConfig> | AuthModuleJwtConfig;
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    inject?: any[];
+    currentUserLoaderService?: Type<CurrentUserLoaderInterface>;
+    currentUserDto: Type<CurrentUser>;
+}
+
 @Module({})
 export class AuthModule {
```

Comment:
Does it make sense to put the two authedUser and jwt implemetnations into one Reusable AuthModule instead of having two?

Or, should we even have a ready AuthModule in the library (instead of the application creating it from scratch) - now that this has been simplified a lot?

Reply:
I can think of these ways:
- One Auth-Module in library supporting two strategies (like this implementation)
- Two Auth-Modules in library each providing a strategy
- Two separate strategies, the module has to be provided by the application

I decided for the first one, which requires minimum code in the application. It has to switch between the environments in two locations:
- Return value in the useFactory-Method of the AuthModule-Registration (https://github.com/vivid-planet/comet/pull/903/files#diff-15be08d5b426bb1584c2baed7828a30bd388a27e9aaf8f42c6d6a3d1553c8b33R70)
- Registration of the global guard (https://github.com/vivid-planet/comet/pull/903/files#diff-32f5b1e5dbfdc4b135ac43e094e31ae167fc8ce304dd5d9da17d7b17519e6356R51)

Reply:
Can you show us how this will look like in a "real" application (or in starter)?

Reply:
> Registration of the global guard

For what reason did you decide against our initial idea to inject a real, static JWT? (That would not require the second guard)

Reply:
Currently the JWT is extracted from a header and validated against a public key which is obtained from a JWKS-Url.
Simulating a JWT would require creating a JWT with a symmetric secret and overriding the explained mechanism of validation (obtain JWT from config, validate againt symmetric key). While this might be technically possible it's not how a JWT is meant to be used.
However, it has the advantage of a sleaker implementation without having to deal with two strategies.

Are there more votes for one or another solution?

Reply:
I changed the implementation to creating a JWT for the local authed user, it indeed simplify things.